### PR TITLE
Modmail added

### PR DESCRIPTION
**Pull request by makian123**

### CODE REVIEW

**Comment by makian123** [comment]
Merge ready

File: src/cogs/modmail.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        embed.set_author(name = user.name, icon_url = user.avatar.url)
+
+        await created_thread.add_user(user)
+        await created_thread.send(
+            f"{user.mention} please wait for <@&{conf.staff_role}> to respond.\n",
+            embeds = [embed]
+        )
+
+        await interaction.response.send_message(
+            f"Channel created in <t:{int(time.time())}:f>\n{created_thread.jump_url}",
```
Comment (Tsche) [inline src/cogs/modmail.py]: why add time to this message?

File: src/cogs/modmail.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        discord.SelectOption(label = "There's an issue with a moderative action"),
+        discord.SelectOption(label = "I want to appeal a ban"),
+        discord.SelectOption(label = "Other")
+    ])
+    async def select_type(self, interaction, selection):
+        await interaction.response.defer(ephemeral = True, thinking = False)
+
+    @discord.ui.button(label = "Create ticket", custom_id = "persistent_view:create_ticket")
+    async def button_callback(self, interaction, button):
+        if len(self.children[0].values) == 0:
```
Comment (Tsche) [inline src/cogs/modmail.py]:
```py
if self.children[0].values:
  ...
```
is sufficient, collections with no elements in them are falsey in Python

**Comment by makian123** [comment]
Well, seems like its all done

File: src/cogs/modmail.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            return
+
+        await interaction.response.send_modal(ModmailModal(self.children[0].values[0]))
+
+class Modmail(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command()
+    async def modmail(self, ctx, channel: discord.TextChannel):
```
Comment (Tsche) [inline src/cogs/modmail.py:74]: this should probably be a hidden Command

```py
@commands.command(hidden=True)
...
```

Comment (Tsche) [reply src/cogs/modmail.py:74]: in theory you can check for the role using a decorator too, but I don't think that's an issue :)

File: src/cogs/modmail.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            invitable = False
+        )
+
+        embed = discord.Embed(title = f"**{self.selected}**")
+        embed.add_field(name="", value=f"{self.issue.value}")
+        embed.set_author(name = user.name, icon_url = user.avatar.url)
+
+        await created_thread.add_user(user)
+        await created_thread.send(
+            f"{user.mention} please wait for <@&{conf.staff_role}> to respond.\n",
```
Comment (Tsche) [inline src/cogs/modmail.py]: please ping admins only for moderative issues

**Review by Tsche — CHANGES_REQUESTED** [review]

File: src/cogs/modmail.py (hunk 1 above)
Comment (makian123) [reply src/cogs/modmail.py]: You requested to show when it was created

Comment (Tsche) [reply src/cogs/modmail.py]: Did I? I think you might have misunderstood me there :P 

Only the link to the created thread should be relevant here

Comment (makian123) [reply src/cogs/modmail.py]: Alr no problem

**Review by Tsche — APPROVED** [review]

File: src/cogs/modmail.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+        created_thread = await channel.create_thread(
+            name = f"{self.name.value}",
+            type = discord.ChannelType.private_thread,
+            invitable = False
+        )
+
+        embed = discord.Embed(title = f"**{self.selected}**")
+        embed.add_field(name="", value=f"{self.issue.value}")
+        embed.set_author(name = user.name, icon_url = user.avatar.url)
```
Comment (Yuhanun) [inline src/cogs/modmail.py:28]: embed = discord.Embed(...)
    .add_field(...)
    .set_author(...)

File: src/cogs/modmail.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+        await created_thread.add_user(user)
+
+        if self.selected == "There's an issue with a moderative action":
```
Comment (Yuhanun) [inline src/cogs/modmail.py:32]: we can't give it like... an ID or something?

**Pull request merged**